pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -218,11 +218,13 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 		_, err := spc.pvcLister.PersistentVolumeClaims(claim.Namespace).Get(claim.Name)
 		switch {
 		case apierrors.IsNotFound(err):
-			_, err := spc.client.Core().PersistentVolumeClaims(claim.Namespace).Create(&claim)
+			created, err := spc.createPersistentVolumeClaim(set, &claim)
 			if err != nil {
 				errs = append(errs, fmt.Errorf("Failed to create PVC %s: %s", claim.Name, err))
 			}
-			spc.recordClaimEvent("create", set, pod, &claim, err)
+			if created || err != nil {
```

Comment:
@ncdc the AlreadyExists error should be returned as part of the error slice so we will force the statefulset to be requeued and pick up the PVC the next time, just the event shouldn't be emitted to users.